packages/@aws-cdk/aws-s3-deployment/lib/source.ts

```diff
@@ -54,7 +56,7 @@ export class Source {
   public static asset(path: string): ISource {
     return {
       bind(context: cdk.Construct): SourceConfig {
-        const asset = new s3_assets.Asset(context, 'Asset', { path });
+        const asset = new s3_assets.Asset(context, `Asset${++uniqueSourceId}`, { path });
```

Comment:
Use the asset path instead of an incrementing ID since the ID depends on the order of sources which can change.